spec/user_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it 'is valid with valid attributes' do
+    user = User.new(name: 'John Doe', posts_counter: 5)
+    expect(user).to be_valid
+  end
+
+  it 'is not valid without a name' do
+    user = User.new(name: nil, posts_counter: 5)
+    expect(user).to_not be_valid
+  end
+
+  it 'is not valid with a negative posts_counter' do
+    user = User.new(name: 'Jane Smith', posts_counter: -1)
+    expect(user).to_not be_valid
+  end
+
+  it 'is valid with zero posts_counter' do
+    user = User.new(name: 'Alice Johnson', posts_counter: 0)
+    expect(user).to be_valid
+  end
```

Comment:
- [ ] Good job so far ✨ However, I noticed that the unit tests for `recent_posts` method present in the `User` model are missing ⚠Similarly the other models' methods are also missing the unit tests. Kindly note that as per [this project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), we need to add unit specs for ALL(`User`, `Post`, `Comment`, and `Like`) of your models' methods ⛏💪


**_Kindly make sure to test ALL of your models' methods (Please apply this to all models)_**

![image](https://github.com/Eccb7/Blog-app/assets/106119105/b1e00a9e-c534-41ef-af9d-729b068a08c5)